syntax = "proto2";

//日常活动项
message DailyActivityData
{
	required uint32 id = 1; //配置id
	required uint32 count = 2; //已完成次数
}

//侠义活动项
message ChivalryActivityData
{
	required uint32 id = 1; //配置id
	required uint32 val = 2; //已获得侠义值
}

//活动信息通知
message ActivityInfoNotify
{
	repeated DailyActivityData dailys = 1; //日常、限时、阵营
	repeated ChivalryActivityData chivalrys = 2; //侠义
	repeated uint32 activeRewardIds = 3; //活跃度已领奖列表
}

//领取活跃度奖励请求
message ActiveRewardRequest
{
	required uint32 rewardId = 1; //奖励id
}

//领取活跃度奖励应答
message ActiveRewardAnswer
{
	required uint32 result = 1; //结果，0：成功，否则失败
	required uint32 rewardId = 2; //奖励id
}

//英雄挑战活动主界面信息应答
message HeroChallengeMainInfoAnswer
{
	required uint32 result = 1; //结果，0：成功，否则失败
	required uint32 all_num = 2; //今日已挑战总次数
	repeated HeroChallengedata data = 3; //详细信息
}

//单个被挑战的英雄数据
message HeroChallengedata
{
	required uint32 hero_id = 1; //id
	required uint32 num = 2; //今日已经挑战次数
	required uint32 star = 3; //历史最高星级
	required uint32 reward_flag = 4; //玩家身上是否有扫荡奖励未领取
}

//英雄挑战扫荡请求
message HeroChallengeSweepRequest
{
	required uint32 hero_id = 1; //英雄id
	required uint32 item_flag = 2; //需要消耗的物品类型 1: 道具 2: 元宝或者绑定元宝
}

//英雄挑战扫荡应答
message HeroChallengeSweepAnswer
{
	required uint32 result = 1; //应答结果 0:成功 其余失败
	required uint32 silver = 2; //获得银两
	required uint32 bind_acer = 3; //获得绑定元宝
	required uint32 acer = 4; //获得元宝
	required uint32 exp = 5; //获得经验
	repeated HeroChallengeSweepItemInfo item_info = 6; //物品信息
}

message HeroChallengeSweepItemInfo
{
	required uint32 item_id = 1; //物品id
	required uint32 item_num = 2; //物品数量
}

//英雄挑战请求领取扫荡奖励，扫荡奖励信息请求共用
message HeroChallengeReciveSweepRewardRequest
{
	required uint32 hero_id = 1; //英雄id
}

//战力达人礼包信息
message ZhanlidarenGiftData
{
	required uint32 id = 1; //配置表ID
	required uint32 getNum = 2; //已领取数量
	required bool   isGet = 3; //是否已领取
}

//战力达人活动信息
message ZhanlidarenActivityData
{
	repeated ZhanlidarenGiftData gifts = 1;
}

//单个限时活动信息
message TimeLimitActivityData
{
	required uint32 activityId = 1; //活动ID
	required uint32 beginTime = 2; //开始时间
	required uint32 endTime = 3; //结束时间
	optional ZhanlidarenActivityData zhanlidaren = 4; //战力达人类型信息
}

//限时活动信息通知
message TimeLimitActivityInfoNotify
{
	repeated TimeLimitActivityData activitys = 1;
}

//限时活动结束通知
message TimeLimitActivityEndNotify
{
	required uint32 activityId = 1;
}

//战力达人礼包数量更新通知
message ZhanlidarenGiftNumNotify
{
	required uint32 activityId = 1;
	required uint32 id = 2; //配置表ID
	required uint32 getNum = 3; //已领取数量
}

//战力达人领取礼包请求
message ZhanlidarenGetRewardRequest
{
	required uint32 activityId = 1; //活动ID
	required uint32 id = 2;
}

//战力达人领取礼包应答
message ZhanlidarenGetRewardAnswer
{
	required uint32 result = 1; //结果，0：成功，否则失败
	required uint32 activityId = 2; //活动ID
	required uint32 id = 3;
}



